draft.md

```diff
@@ -0,0 +1,99 @@
+# Draft IPLD Unixfs Spec
+
+## Basic Structure
+
+A Unixfs is either a file or a directory.
+The top level IPLD object is a CBOR map with at least two fields: `type` and `data`
+and maybe a few other such as a version string or a set of flags.
+The `type` field is either `file` or `dir`.
+
+## IPLD `file`
+
+If an IPLD file is a leaf its CID type is `raw` (0x55) and has no structure.
+Otherwise its CID type is `dag-cbor` (0x71).
+The `type` field is set to `file` and the `data` field is an CBOR array.
+Each element of the array is CBOR map with the following fields:
+
+  - `data`: link
+  - `size`: cumulative size of `data`
+  - `fsize`: (file size) cumulative size of the payload of `data`
```

Comment:
@kevina this seems backwards. The `size` of the protocol-wrapped objects is currently [optional for all intents and purposes](https://github.com/ipfs/go-ipfs/issues/4286#issuecomment-336195370). On the other hand the actual `fsize` is mandatory if what you are expressing is a "top node of a file DAG".

Reply:
To be clear `fsize` is the logical bytes while `size` is the direct or size that included the overhead of interior nodes. `fsize <= size`

Reply:
@kevina precisely. The logical bytes ( `fsize` ) are interesting because they are needed to calculate buffers / HTTP header / etc

The node overhead is only useful for estimating how much local storage would be necessary to grab the entire DAG locally, but given the overhead is never more than ~10 bytes per node, it becomes ( from my PoV at least ) needless cruft.

I've done a lot of tests over the last year with DAGs specifically excluding mention of what you refer to as `size`. Everything works fine. I strongly believe the overhead-including-size should not be a mandatory part of any future spec ( and thus should be the last item in the CBOR array, not an intermediate one )

Reply:
>  I strongly believe the overhead-including-size should not be a mandatory part of any future spec 

Note: This is not a CBOR array but a map the structure is `{type: "file", data: [{/*link*/}, {}, {} ...]}`.

I don't have a strong opinion on which sizes to include.  @whyrusleeping thoughts?

Reply:
I think `size` and `fsize` are both useful - download progress would depend on `size`, for example, as you still need to fetch the wrapper bytes.